apps/src/templates/studioHomepages/teacherHomepageV2/CourseContentDropdown.tsx

```diff
@@ -1,24 +1,104 @@
+import {SimpleDropdown} from '@code-dot-org/component-library/dropdown';
 import {BodyThreeText} from '@code-dot-org/component-library/typography';
-import React from 'react';
+import React, {useEffect, useState, useMemo} from 'react';
+import {useNavigate} from 'react-router-dom';
 
 import {Section} from '@cdo/apps/templates/teacherDashboard/types/teacherSectionTypes';
+import HttpClient from '@cdo/apps/util/HttpClient';
 import i18n from '@cdo/locale';
 
+import {TEACHER_NAVIGATION_SECTIONS_URL} from '../../teacherNavigation/TeacherNavigationPaths';
+
+import {TaskButton} from './TaskButton';
+
 import styles from './teacherHomepage.module.scss';
 
 interface CourseContentDropdownProps {
   section: Section;
 }
 
+interface UnitLessons {
+  value: string;
+  text: string;
+}
+
+/**
+ * CourseContentDropdown component.
+ * Used to render a dropdown for selecting a lesson to navigate to.
+ * @param section - Section object containing the course display name.
+ */
 export const CourseContentDropdown: React.FC<CourseContentDropdownProps> = ({
   section,
 }) => {
+  const navigate = useNavigate();
+  const [lessonList, setLessonList] = useState<UnitLessons[]>([]);
+
+  // Retrieve units and lessons for the section
+  useEffect(() => {
+    if (section.unitId) {
+      HttpClient.fetchJson<UnitLessons[]>(
+        `/sections/retrieve_lessons_for_dropdown/${section.id}`
```

Comment:
with the routing change requested, this will need to change to `sections/${section.id}/retrieve_lessons_for_dropdown`